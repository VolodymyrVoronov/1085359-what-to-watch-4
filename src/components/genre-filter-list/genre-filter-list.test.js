import React from "react";
import {Provider} from "react-redux";
import renderer from "react-test-renderer";
import configureStore from "redux-mock-store";
import GenreFilterList from "./genre-filter-list.jsx";

const GENRES = [`Comedy`, `Sci-Fi`, `Horror`];

const mockStore = configureStore([]);

it(`render should be match markup`, () => {

  const store = mockStore({
    genreFilterIndex: 0,
  });

  const result = renderer
    .create(<Provider store={store}>
      <GenreFilterList
        genres={GENRES}
        currentGenre={GENRES[0]}
      />
    </Provider>)
    .toJSON();

  expect(result).toMatchSnapshot();
});
