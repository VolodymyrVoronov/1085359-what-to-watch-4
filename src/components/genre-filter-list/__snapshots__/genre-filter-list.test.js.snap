// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render should be match markup 1`] = `
<ul
  className="catalog__genres-list"
>
  <li
    className="catalog__genres-item catalog__genres-item--active"
  >
    <a
      className="catalog__genres-link"
      href="#"
      onClick={[Function]}
    >
      Comedy
    </a>
  </li>
  <li
    className="catalog__genres-item"
  >
    <a
      className="catalog__genres-link"
      href="#"
      onClick={[Function]}
    >
      Sci-Fi
    </a>
  </li>
  <li
    className="catalog__genres-item"
  >
    <a
      className="catalog__genres-link"
      href="#"
      onClick={[Function]}
    >
      Horror
    </a>
  </li>
</ul>
`;
